astropy/nddata/bitmask.py

```diff
@@ -50,11 +50,43 @@ def _is_int(n):
     )
 
 
+class NotPowerOf2(ValueError):
+    """ Indicates that a value is not an integer that is a power of 2. """
+    pass
+
+
+class BitFlag(int):
+    """ Bit flags: integer values that are powers of 2. """
+    def __new__(cls, val, doc=None):
+        if isinstance(val, tuple):
+            if doc is not None:
+                raise ValueError("Flag's doc string cannot be provided twice.")
+            val, doc = val
+
+        if not _is_int(val):
+            raise TypeError("Flag values must be of integral type.")
+
+        if not _is_bit_flag(val):
+            raise NotPowerOf2("Value {} is not a valid bit flag: bit flag"
+                              "values must be powers of two.".format(val))
+
```

Comment:
Seems to me that the first error message is too loose. Might as well say that the value must be an integral power of two right off the bat, and combine both tests into one.